src/controllers/expenses.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const expenseService = require('../services/expenses');
+const userService = require('../services/users');
+
+const getAll = (req, res) => {
+  const {
+    userId,
+    from,
+    to,
+    categories,
+  } = req.query;
+  let expenses = expenseService.getAll();
+
+  if (+userId) {
+    expenses = expenses.filter(expense => expense.userId === +userId);
+  }
+
+  if (from) {
+    expenses = expenses.filter(expense => expense.spentAt >= from);
+  }
+
+  if (to) {
+    expenses = expenses.filter(expense => expense.spentAt <= to);
+  }
+
+  if (categories) {
+    expenses
+      = expenses.filter(expense => categories.includes(expense.category));
+  }
```

Comment:
This logic belongs to service, not the controller. Move all filtering/search logic there.

In a real life scenario, `.getAll` would do some database call inside. Downloading the entire dataset and then filtering in your controller code would be inefficient, so you should just pass your filtering options to service and do your filtering there.